R/utilities-data-combined.R

```diff
@@ -0,0 +1,164 @@
+# TODO: move these functions to the utilities package
+
+#' Count number of objects
+#'
+#' @details
+#'
+#' If the argument is not a vector, unlike `length()`, this function will not
+#' count the number of named bindings in an environment, but only the number of
+#' instances of a class.
+#'
+#' For example, `length(mtcars)` will return 11, but `objCount(mtcars)` will
+#' return 1.
+#'
+#' @param x An object (an atomic vector, a list, or instance(s) of a class).
+#'
+#' @examples
+#'
+#' objCount(c(1, 2, 3)) # 3
+#' objCount(list("a", "b")) # 2
+#' objCount(mtcars) # 1
+#'
+#' @return Integer representing the count of objects.
+#'
+#' @keywords internal
+#' @noRd
+
+objCount <- function(x) {
+  # `is.vector()` can handle both atomic vectors and lists, i.e.
+  # both `is.vector(c(1, 2))` and `is.vector(list(1, 2))` will be `TRUE`
+  if (is.vector(x)) {
+    l <- length(x)
+  } else {
+    l <- length(list(x))
```

Comment:
I am wondering about this. SInce x is not a vector, list(x) will have one item no? Isn't the value always 1?
Code can be simplified by returning early

Reply:
Good point. Done!